### ISSUE TITLE
No JSON Data

### ISSUE THREAD

**Issue by JordanSicherman** [shusain93/OSXMessageProxy#14]
When double clicking the StartIMProxy.command, the server starts up as expected and I can connect to it in AndromedaB, but no JSON is displayed in the server console and AndromedaB complains that no JSON data is available.

I'm using a 2011 Mac Air on Sierra and Galaxy S7 running the latest Android OS with the latest OSXMessageProxy and AndromedaB installations.

**Comment by shusain93** [comment]
Do you get nothing at all in the logs? When you say the app connects do you mean you see the banner pop up that says connected or whatever? It could be that your message database can't be read for whatever reason

**Comment by JordanSicherman** [comment]
When I click the startup script, I get this in the console and nothing else.

jordan:~ Jordan$ /Users/Jordan/Desktop/MessageProxy/StartIMProxy.command ; exit;

**Comment by JordanSicherman** [comment]
To summarize, after playing around with it for quite awhile:

1. I run the StartIMProxy.command by double clicking.
2. It makes a log folder, logs "Starting Server at ____" and displays nothing in the terminal.
3. OSXMessageProxy opens and says "Ready at _____".
4. I connect to the server from my Android and a banner shows up that says Connected.
5. A few seconds later, a banner appears on my Android that says "Unable to load conversationJSONDatab..s" (can't see the full banner on my phone).

If I browse to my.domain:8735/conversations?t=password, I can see a bunch of message JSON data so it's being loaded... Perhaps it's crashing because my contacts have emojis in their name?

**Comment by JordanSicherman** [comment]
This is actually an issue with AndromedaB. It is simply timing out before it receives the list of conversations (I have a lot. Like, 194000). Fixed by the addition of
`stringRequester.setRetryPolicy(new DefaultRetryPolicy(500000,
                DefaultRetryPolicy.DEFAULT_MAX_RETRIES,
                DefaultRetryPolicy.DEFAULT_BACKOFF_MULT));`
to line 45 of RemoteMessagesInterface#getConversations()

**Comment by shusain93** [comment]
Yeah I'm going to say "wont-fix" to that issue haha. Guess I'm not popular enough to need that? I'd look at getting rid of some conversations because the conversation list is sent on every new message as needed which would mean a ton of useless data is sent if you have 194k conversations. I should limit it though to maybe only returning the 50 most recent conversations.

**Comment by JordanSicherman** [comment]
Ahaha yeah good plan. I've been working on some mods in an offline copy of Andromeda... I've got a compose message setup (had to add a /contacts command to the server to simplify) running and I weeded off any big conversations.

Something else I've come across is that the JSON (and therefore andromeda) will distinguish between one contact with two contact options (ie. email and phone) where iMessage will condense them into one conversation. This causes an issue to arise in some compositions.

I'll probably fork your repo and submit some PRs when I have reliable and tidy changes. Great project!
